src/main/java/bio/terra/cloudfiletodatastore/deltalayer/functions/DeltaLayerBackgroundFunction.java

```diff
@@ -0,0 +1,36 @@
+package bio.terra.cloudfiletodatastore.deltalayer.functions;
+
+import bio.terra.cloudevents.GCSEvent;
+import bio.terra.cloudfiletodatastore.FileUploadedMessage;
+import bio.terra.cloudfiletodatastore.deltalayer.DeltaLayerFileUploadedMessageProcessor;
+import com.google.cloud.functions.BackgroundFunction;
+import com.google.cloud.functions.Context;
+import java.time.ZoneOffset;
+import java.util.logging.Logger;
+
+/**
+ * Implementation of a GCF background function
+ * https://cloud.google.com/functions/docs/writing#background_functions This would be the ideal if
+ * not for the crummy Gson support which prevents us currently from using a Google maintained class
+ * to capture the event data.
+ */
```

Comment:
does the crummy gson support throw an error if our copy of `GCSEvent` leaves out all the fields we don't care about? For instance, if we removed `kmsKeyName` from `GCSEvent`, would it cause a problem? I like the `DeltaLayerBackgroundFunction` approach better than the other two, IF we can keep our maintenance of `GCSEvent` low.

Reply:
Gson seems to be very forgiving.  Extra data in the JSON payload is ignored, missing data is deserialized as null.  Thus, there are fewer parse errors, but greater potential for downstream NPEs.  I've added a GcsEvent deserialization test confirming the expected behavior.

Reply:
The `DeltaLayerBackgroundFunction` can just extend a `CloudStorageEventHarness<GCSEvent>` and put all the business logic in the `doAccept()` method. [QA-1544](https://github.com/DataBiosphere/terra-gcs-bq-streaming-functions/pull/11) has the `CloudStorageEventHarness`

`public abstract class CloudStorageEventHarness<T> implements BackgroundFunction<T>`

Reply:
I don't see the advantage of adding the CloudStorageEventHarness as a superclass.   Am I missing something?